src/poetry/repositories/pool.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+import warnings
+
+from typing import TYPE_CHECKING
+
+from poetry.repositories.repository_pool import RepositoryPool
+
+
+if TYPE_CHECKING:
+    from poetry.repositories.repository import Repository
+
+
+class Pool(RepositoryPool):
+    def __init__(
+        self,
+        repositories: list[Repository] | None = None,
+        ignore_repository_names: bool = False,
+    ) -> None:
+        warnings.warn(
+            "Object Pool from poetry.repositories.pool is renamed and scheduled for"
+            " removal in poetry release 1.4.0. Please migrate to RepositoryPool from"
+            " poetry.repositories.repository_pool.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
```

Comment:
IMO the deprecation warning should be at module level because not only the class but the module is deprecated.

Reply:
I would agree, but since `poetry.repositories.pool` is also imported in `poetry.repositories.__init__`, this will raise the warning at any `poetry.repositories(.*)` import. As a user I would only expect to see a `DeprecationWarning` if I import the thing that is deprecated, so therefore I opted for this suboptimal approach (also: no warning for typing with deprected `Pool`). 

The rationale for importing `Pool` in `__init__.py` is that `poetry-plugin-import` depends on it (imports `Pool` from poetry.repositories`).